src/python_framework_v02/troute/nhd_network.py

```diff
@@ -417,3 +423,92 @@ def replace_waterbodies_connections(connections, waterbodies):
             # copy to new network unchanged
             new_conn[n] = connections[n]
     return new_conn
+
+
+def build_subnetworks(connections, rconn, min_size, sources=None):
+    """
+    Construct subnetworks using a truncated breadth-first-search
+    
+    Arguments:
+        connections
+        rconn
+        max_depth
+        sources
+    Returns:
+        subnetwork_master
+    """
+    # if no sources provided, use tailwaters
+    if sources is None:
+        # identify tailwaters
+        sources = rconn.keys() - chain.from_iterable(rconn.values())
```

Comment:
Is there a reason `headwaters(rconn)` isn't being called here?

Reply:
resolving with re-written functions